### DISCUSSION TITLE
Should vcpkg use a FATAL_ERROR clause at the end of the `cmake_minimum_required` stmt?

### DISCUSSION THREAD

**Discussion by stephengtuggy** [microsoft/vcpkg#25657] [Ideas]
According to https://cmake.org/cmake/help/latest/command/cmake_minimum_required.html :

> The FATAL_ERROR option is accepted but ignored by CMake 2.6 and higher. It should be specified so CMake versions 2.4 and lower fail with an error instead of just a warning.

So I would suggest that the `cmake_minimum_required` line at the top of vcpkg's CMakeLists.txt files, should use `FATAL_ERROR`. Just in case someone tries to use them with CMake 2.4 or earlier. (Probably a remote chance, but still.)

Like this:

```cmake
cmake_minimum_required(VERSION 3.9 FATAL_ERROR)
```

**Comment by Osyotr** [comment]
Minimum cmake version required to use vcpkg is 3.7.2
https://github.com/microsoft/vcpkg/pull/23134
